Source/Krypton Toolkit/Krypton.Toolkit.Suite.Extended.CheckSum.Tools/UX/KryptonComputeFileCheckSum.cs

```diff
@@ -794,23 +794,68 @@ private void SaveHashToFile(string text, bool toUpper)
                 }
                 else if (kcmbAlgorithimType.Text == "SHA-1")
                 {
-                    bgSHA1Hash.RunWorkerAsync(ktxtFilePath.Text);
+                    dlg.Filter = "SHA-1 Hash Files|*.sha1";
```

Comment:
This is screaming for a switch statement, and then call the ShowDialog afterwards

Reply:
I'll have a look tomorrow

Reply:
Has this been done ?

Reply:
Just doing it now, will be done by tomorrow

Reply:
Did this get done?
Did I miss the change ?

Reply:
Now fixed, put the logic into separate methods